src/e3/anod/qualifiers_manager.py

```diff
@@ -0,0 +1,844 @@
+            # * repr_omit_key
+
+            if qualifier["repr_omit_key"]:
+                return "_" + qualifier_value
+            else:
+                return "_" + qualifier_name + "-" + qualifier_value
+        else:
+            raise AnodError(
+                "An expected qualifier type was encountered during parsing "
+                f'Got "{qualifier["type"]}".'
+            )
+
+    @property
+    def component(self) -> str | None:
+        """Return the component name.
+
+        If the current qualifier configuration conrespond to a component name,
+        then return this name, else return None.
+
+        This function cannot be called before the end of the declaration phase
+        (i.e. before the call to parse)
+
+        :return: The component name.
+        """
+        if not self.is_declaration_phase_finished:
+            raise AnodError(
+                "It is not possible to build a component before the end of the"
+                "declaration phase."
+            )
+
+        if self.component_names is None:
+            raise AnodError(
+                "The parse method must be called first to generate the components "
+                "names."
+            )
+
+        return self.component_names.get(self.__key(self.anod_instance.parsed_qualifier))
+
+    @property
+    def build_space_name(self) -> str:
+        """Return the build space name.
+
+        Aggregate the build space name. It is made of four parts:
+        base + qualifier_suffix + hash + test
+
+         * base: The component_prefix or, by default, the spec name.
+         * qualifier_suffix: The concatenation of the contribution of
+            all qualifiers.
+         * hash: The hash of the aggregation of all qualifiers marked with
+            'repr_in_hash'.
+         * test: '_test' if the current anod primitive is test.
+
+        If the generated component is not None, then use it for consistency reason.
+
+        This function cannot be called before the end of the declaration phase
+        (i.e. before the call to parse)
+
+        :return: The build_space_name.
+        """
+        if not self.is_declaration_phase_finished:
+            raise AnodError(
+                "It is not possible to build a build_space_name before the end of the"
+                "declaration phase."
+            )
+
+        component_name = self.component
+
+        if component_name is not None:
+
+            # A component name has been defined and must be used.
+            return component_name
+
+        parsed_qualifiers = self.__get_parsed_qualifiers()
+
+        qualifier_suffix = ""
+        hash_suffix = ""
+        kind_suffix = "_test" if self.anod_instance.kind == "test" else ""
+
+        qualifier_names_list = list(self.qualifier_decls.keys())
+
+        # Reset the hash_pool to let __generate_qualifier_part populates it
+        self.hash_pool = ""
+
+        # Ensure reproducibility
+        qualifier_names_list.sort()
```

Comment:
No need to sort in place here:

for qualifier_name in sorted(qualifier_names_list):...